packages/discovery-provider/ddl/migrations/0045_add_track_metadata_columns.sql

```diff
@@ -0,0 +1,50 @@
+begin;
+
+
+-- add new columns
+alter table tracks add column if not exists is_downloadable boolean not null default false;
+alter table tracks add column if not exists is_download_gated boolean not null default false;
+alter table tracks add column if not exists download_conditions jsonb;
+alter table tracks add column if not exists is_original_available boolean not null default false;
+alter table tracks add column if not exists orig_file_cid varchar;
+alter table tracks add column if not exists orig_filename varchar;
+
+
+-- set new downloadable column to true for donwloadable tracks
+update tracks set is_downloadable = true where is_current is true and (download->>'is_downloadable')::boolean is true;
+
+
+-- set track cid for downloadable tracks whose track cid columns are missing values
+update tracks t set track_cid = download_cid.cid
+from (
+  select track_id, download->>'cid' as cid from tracks
+  where is_current is true
+  and track_cid is null
+  and (download->>'cid')::text != ''
+) as download_cid
+where t.is_current is true
+and t.track_id = download_cid.track_id;
+
+
+-- inherit download conditions from premium conditions for stream gated downloadable tracks
+update tracks set is_download_gated = true
+  where is_current is true and is_premium is true and (download->>'is_downloadable')::boolean is true;
+update tracks set download_conditions = premium_conditions
+  where is_current is true and is_premium is true and (download->>'is_downloadable')::boolean is true;
+
+
+-- set follow gated download conditions for non gated tracks that are downloadable and that require follow
+update tracks t
+set is_premium = true, download_conditions = jsonb_build_object('follow_user_id', download_requires_follow.owner_id)
```

Comment:
shouldn't this be `is_download_gated = true` not `is_premium = true`?

Reply:
lol yup, my bad